src/vos/vos_obj.h

```diff
@@ -73,18 +76,29 @@ struct vos_object {
  * not in cache, this function will load it from PMEM pool or create it, then
  * add it to the cache.
  *
- * \param occ	[IN]	Object cache, it could be a percpu data structure.
- * \param cont	[IN]	Open container.
- * \param oid	[IN]	VOS object ID.
- * \param no_create [IN]
- *			Do not allocate object if it's not there yet.
- * \param intent [IN]	The request intent.
- * \param obj_p [OUT]	Returned object cache reference.
+ * \param occ		[IN]		Object cache, can be per cpu
+ * \param cont		[IN]		Open container.
+ * \param oid		[IN]		VOS object ID.
+ * \param epr		[IN,OUT]	Epoch range.   High epoch should be set
+ *					to requested epoch.   The lower epoch
+ *					can be 0 or bounded.
+ * \param no_create	[IN]		If object doesn't exist, do not create
+ * \param intent	[IN]		The request intent.
+ * \param visible_only	[IN]		Return the object only if it's visible
+ * \param obj_p		[OUT]		Returned object cache reference.
+ *
+ * \return	0			The object is visible or, if
+ *					\p visible_only is false, it has
+ *					punched data or is entirely empty.
+ * \return	-DER_NONEXIST		The conditions for success don't apply
+ *		-DER_INPROGRESS		The local target doesn't have the
+ *					definitive state of the object.
+ *		other			Another error occurred
  */
 int
 vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
-	     daos_unit_oid_t oid, daos_epoch_t epoch,
-	     bool no_create, uint32_t intent, struct vos_object **obj_p);
+	     daos_unit_oid_t oid, daos_epoch_range_t *epr, bool no_create,
```

Comment:
I think we can convert "intent" to "no_create", and remove "no_create" to make the API cleaner